src/Serializer/ItemNormalizer.php

```diff
@@ -37,11 +39,17 @@ public function normalize($object, $format = null, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
         if (isset($data['id']) && !isset($context['object_to_populate'])) {
+            if (isset($context['allow_update']) && true !== $context['allow_update']) {
```

Comment:
> I'm afraid that it's too late for this particular change in rc3.
> It will break almost all apps already using API Platform 2

This is exactly why we should do it now, while we can still afford to break things. Minor inconvenience for those who are using a non-stable release, if you ask me.

> especially because we also maintain the Symfony Serializer and we can prevent problems in Symfony core

Abuse of power :stuck_out_tongue: jk

> Prefixing context key will also make the config of all apps using API Platform uglier and harder to remain.

It's a necessary evil. The alternative is worse.